errors/errors.go

```diff
@@ -1,409 +1,53 @@
-		msg:       message,
-		CallStack: stack.New(1),
-	}
+// Is reports whether any error in err's chain matches target.
+func Is(err, target error) bool {
+	return errors.Is(err, target)
 }
 
-// Errorf formats according to a format specifier and returns the string
-// as a value that satisfies error.
-// Errorf also records the stack trace at the point it was called.
-func Errorf(format string, args ...interface{}) error {
-	return &fundamental{
-		msg:       fmt.Sprintf(format, args...),
-		CallStack: stack.New(1),
-	}
+// Errorf is deprecated use fmt.Errorf instead
+func Errorf(format string, a ...any) error {
+	return fmt.Errorf(format, a...)
 }
 
-// fundamental is an error that has a message and a stack, but no caller.
-type fundamental struct {
-	*stack.CallStack
-	msg string
+// As finds the first error in err's chain that matches target, and if so, sets
+// target to that error value and returns true.
+func As(err error, target any) bool {
+	return errors.As(err, target)
 }
 
-func (f *fundamental) Error() string { return f.msg }
-
-func (f *fundamental) Format(s fmt.State, verb rune) {
-	switch verb {
-	case 'v':
-		if s.Flag('+') {
-			_, _ = io.WriteString(s, f.msg)
-			f.CallStack.Format(s, verb)
-			return
-		}
-		fallthrough
-	case 's':
-		_, _ = io.WriteString(s, f.msg)
-	case 'q':
-		fmt.Fprintf(s, "%q", f.msg)
-	}
+// New returns an error that formats as the given text.
+// Each call to New returns a distinct error value even if the text is identical.
+func New(text string) error {
+	return errors.New(text)
 }
 
-// WithStack annotates err with a stack trace at the point WithStack was called.
-// If err is nil, WithStack returns nil.
-func WithStack(err error) error {
-	if err == nil {
-		return nil
-	}
-	return &withStack{
-		err,
-		stack.New(1),
-	}
-}
-
-type withStack struct {
-	error
-	*stack.CallStack
-}
-
-func (w *withStack) Cause() error  { return w.error }
-func (w *withStack) Unwrap() error { return w.error }
-func (w *withStack) Context() map[string]interface{} {
-	if child, ok := w.error.(HasContext); ok {
-		return child.Context()
-	}
-	return nil
-}
-
-func (w *withStack) Format(s fmt.State, verb rune) {
-	switch verb {
-	case 'v':
-		if s.Flag('+') {
-			fmt.Fprintf(s, "%+v", w.Cause())
-			w.CallStack.Format(s, verb)
-			return
-		}
-		fallthrough
-	case 's':
-		_, _ = io.WriteString(s, w.Error())
-	case 'q':
-		fmt.Fprintf(s, "%q", w.Error())
-	}
+// Unwrap returns the result of calling the Unwrap method on err, if err's
+// type contains an Unwrap method returning error.
+// Otherwise, Unwrap returns nil.
+func Unwrap(err error) error {
+	return errors.Unwrap(err)
 }
 
-// WrapStack returns an error annotating err with a stack trace
-// at the depth indicated. A calling depth of 1 is the function that
-// called WrapStack. If err is nil, Wrap returns nil.
-func WrapWithDepth(err error, message string, depth int) error {
+// Wrap is deprecated use fmt.Errorf() instead
+func Wrap(err error, msg string) error {
```

Comment:
I believe we should keep `Wrap` an `Wrapf` because they add stack. And `fmt.Errorf` does not. And also this should be reverted to keep storing stack.